loopback/loopback.go

```diff
@@ -0,0 +1,539 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+// +build darwin
```

Comment:
There are some macOS specific stuff that won't build on Linux. But not a lot so I'll make Linux-specific ones today